Assets/Scripts/UI/MainMenu/MainMenuTumbleweed.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Unity.Mathematics;
+using UnityEngine;
+
+public class MainMenuTumbleweed : MonoBehaviour
+{
+    [SerializeField]
+    private Transform target;
+    [SerializeField]
+    private GameObject model;
+    [SerializeField]
+    private Terrain terrain;
+    private float terrainHeight;
+
+    private void Start()
+    {
+        // Rotates model 360 degrees 
+        model.transform.LeanRotateAroundLocal(model.transform.forward,360,1).setLoopCount(-1);   
+
+
+    }
+
+    private void LateUpdate()
+    {
+        // Get the height of the terrain at the position of TumbleweedParent
+        terrainHeight = terrain.SampleHeight(transform.position);
```

Comment:
I have suspicion that running this function every update is pretty costly (depending on if Unity samples their terrain from a heigthmap of some sort or by actually calculating geometry). 
Might not be a problem as there isn't much other perfomance heavy stuff at all in the main menu, but we might have to monitor this a bit

Reply:
Since the terrain is created by painting on a heightmap, this should devolve down to sampling that texture and thus perform well.